src/main/scala/org/broadinstitute/dsde/workbench/sam/google/DisableUsersMonitor.scala

```diff
@@ -0,0 +1,192 @@
+  final case class DisableUserResponse(userId: WorkbenchUserId, value: Option[UserStatus])
+  final case class ReportMessage(value: DisableUserResponse, ackId: String) extends DisableUserResult(ackId = ackId)
+  final case class FailToDisable(t: Throwable, ackId: String) extends DisableUserResult(ackId = ackId)
+
+  def props(
+             pollInterval: FiniteDuration,
+             pollIntervalJitter: FiniteDuration,
+             pubSubDao: GooglePubSubDAO,
+             pubSubSubscriptionName: String,
+             userService: UserService
+           ): Props =
+    Props(new DisableUsersMonitorActor(pollInterval, pollIntervalJitter, pubSubDao, pubSubSubscriptionName, userService))
+}
+
+class DisableUsersMonitorActor(
+  val pollInterval: FiniteDuration,
+  pollIntervalJitter: FiniteDuration,
+  pubSubDao: GooglePubSubDAO,
+  pubSubSubscriptionName: String,
+  userService: UserService
+) extends Actor with LazyLogging with FutureSupport {
+  import DisableUsersMonitor._
+  import context._
+
+  self ! StartMonitorPass
+
+  // fail safe in case this actor is idle too long but not too fast (1 second lower limit)
+  setReceiveTimeout(max((pollInterval + pollIntervalJitter) * 10, 1 second))
+
+  private def max(durations: FiniteDuration*): FiniteDuration = {
+    implicit val finiteDurationIsOrdered: FiniteDuration.FiniteDurationIsOrdered.type = scala.concurrent.duration.FiniteDuration.FiniteDurationIsOrdered
+    durations.max
+  }
+
+  override def receive: Receive = {
+    case StartMonitorPass =>
+      // start the process by pulling a message and sending it back to self
+      pubSubDao.pullMessages(pubSubSubscriptionName, 1).map(_.headOption) pipeTo self
+
+    case Some(message: PubSubMessage) =>
+      attemptToDisableUser(message) pipeTo self
+
+    case None =>
+      // there was no message to wait and try again
+      val nextTime = org.broadinstitute.dsde.workbench.util.addJitter(pollInterval, pollIntervalJitter)
+      system.scheduler.scheduleOnce(nextTime.asInstanceOf[FiniteDuration], self, StartMonitorPass)
+
+    case ReportMessage(disableUserResponse, ackId) =>
+      handleDisableUserResponse(disableUserResponse, ackId) pipeTo self
+
+    case FailToDisable(t, ackId) =>
+      t match {
+        case userNotFound: WorkbenchExceptionWithErrorReport if userNotFound.errorReport.statusCode.contains(StatusCodes.NotFound) =>
+          // this can happen if a user is deleted before the disable message is handled
+          // acknowledge it so we don't have to handle it again
+          acknowledgeMessage(ackId).map(_ => StartMonitorPass) pipeTo self
+          logger.info(s"user to disable not found: ${userNotFound.errorReport}")
+
+        case regrets: Throwable => throw regrets
+      }
+
+    case Status.Failure(t) => throw t
+
+    case ReceiveTimeout =>
+      throw new WorkbenchException("DisableUsersMonitorActor has received no messages for too long")
+
+    case x => logger.info(s"unhandled $x")
```

Comment:
I would make this message a little more descriptive, maybe: 

```suggestion
    case x => logger.info(s"Received unhandleable message in DisableUsersMonitor: $x")
```

Reply:
actually doing `$x` is a big no no because it writes only the exception message and loses critical information: the exception class and stack trace.

`logger.info(s"Received unhandleable message in DisableUsersMonitor", x)` is correct